src/components/BankTransactions/BankTransactionsHeader.tsx

```diff
@@ -148,30 +152,84 @@ export const BankTransactionsHeader = ({
   const { count } = useCountSelectedIds()
   const { clearSelection } = useBulkSelectionActions()
 
+  const [isConfirmAllModalOpen, setIsConfirmAllModalOpen] = useState(false)
+  const [isCategorizeAllModalOpen, setIsCategorizeAllModalOpen] = useState(false)
+  const [selectedCategory, setSelectedCategory] = useState<CategoryOption | undefined>(undefined)
+
+  const handleConfirmAllClick = useCallback(() => {
+    setIsConfirmAllModalOpen(true)
+  }, [])
+
+  const handleCategorizeAllClick = useCallback(() => {
+    setIsCategorizeAllModalOpen(true)
+  }, [])
+
+  const handleCategorizeModalClose = useCallback((isOpen: boolean) => {
+    setIsCategorizeAllModalOpen(isOpen)
+    if (!isOpen) {
+      setSelectedCategory(undefined)
+    }
+  }, [])
+
   const BulkActions = useCallback(() => {
     return (
       <HStack align='center' gap='sm'>
         <Button
           variant='outlined'
-          // TODO: onClick
+          onClick={handleConfirmAllClick}
         >
-          Categorize
+          Confirm All
         </Button>
+        <BulkActionsConfirmationModal
+          isOpen={isConfirmAllModalOpen}
+          onOpenChange={setIsConfirmAllModalOpen}
+          itemCount={count}
+          actionLabel='confirm'
+          itemLabel='suggestions'
+          onConfirm={() => {}}
+          confirmLabel='Confirm All'
+          cancelLabel='Cancel'
+        />
         <Button
           variant='outlined'
-          // TODO: onClick
+          onClick={handleCategorizeAllClick}
         >
-          Uncategorize
+          Categorize All
         </Button>
-        <Button
-          variant='outlined'
-          // TODO: onClick
+        <BulkActionsConfirmationModal
+          isOpen={isCategorizeAllModalOpen}
+          onOpenChange={handleCategorizeModalClose}
+          itemCount={count}
+          actionLabel='categorize'
+          itemLabel='transactions'
+          descriptionLabel={selectedCategory?.payload?.display_name ? ` as ${selectedCategory?.payload?.display_name}` : ''}
+          onConfirm={() => {}}
+          confirmLabel='Categorize All'
+          cancelLabel='Cancel'
+          confirmDisabled={!selectedCategory}
+          hideDescription={!selectedCategory}
         >
-          Match
-        </Button>
+          <VStack gap='xs'>
+            <Label>Select category</Label>
+            <CategorySelect
+              name='bulk-category-select'
+              value={selectedCategory}
+              onChange={setSelectedCategory}
+              showTooltips={false}
+              excludeMatches={true}
+            />
```

Comment:
We'll need this to point the label at the select with the `htmlFor` prop. Let's also do a `useId` for the name rather than a hardcoded string (see `InvoiceTermsComboBox`)
```suggestion
            <Label htmlFor='bulk-category-select' size='sm'>Select category</Label>
            <CategorySelect
              name='bulk-category-select'
              value={selectedCategory}
              onChange={setSelectedCategory}
              showTooltips={false}
              excludeMatches={true}
            />
```